packages/notifi-core/lib/models/NotificationHistory.ts

```diff
@@ -68,105 +69,68 @@ export type AccountBalanceChangedEventDetails = {
-  tenantName: string;
-  description: string;
-  state?: string;
-  daoUrl?: string;
-  proposalUrl?: string;
-  proposalTitle?: string;
-};
-
-export type DirectTenantMessageEventDetails = {
-  __typename: 'DirectTenantMessageEventDetails';
-  tenantName: string;
-};
-
-export type GenericEventDetails = {
-  __typename: 'GenericEventDetails';
-  sourceName: string;
-  notificationTypeName: string;
-  genericMessage: string;
-  action?: GenericEventAction;
-  icon: string;
-};
-
-export type GenericEventAction = {
-  __typename?: 'GenericEventAction';
-  name: string;
-  url: string;
-};
-
-export type HealthValueOverThresholdEventDetails = {
-  __typename: 'HealthValueOverThresholdEventDetails';
-  name: string;
-  value: string;
-  threshold: string;
-  url: string;
-};
-
-export type NftAuctionChangedEventDetails = {
-  __typename: 'NftAuctionChangedEventDetails';
-  auctionUrl?: string;
-  walletBlockchain: WalletBlockchain;
-  highBidAmount: number;
-  highBidSymbol?: string;
-  imageUrl?: string;
-  auctionTitle: string;
-};
-
-export type NftCollectionsReportEventDetails = {
-  __typename: 'NftCollectionsReportEventDetails';
-  type: NftCollectionsReportType;
-  providerName: string;
-  sourceLink: string;
-  collections: NftCollectionStats[];
-};
-
-export type NftCollectionStats = {
-  __typename?: 'NftCollectionStats';
-  collectionId: string;
-  name: string;
-  imgUrl?: string;
-  volume1Day?: string;
-  volume1DayChange?: string;
-};
-
-export type WalletsActivityReportEventDetails = {
-  __typename: 'WalletsActivityReportEventDetails';
-  providerName: string;
-  sourceLink: string;
-  walletActivityType: WalletsActivityReportType;
-  wallets: WalletsActivityReportWallet[];
-};
-
-export type WalletsActivityReportWallet = {
-  __typename?: 'WalletsActivityReportWallet';
-  address: string;
-  volume1Day: string;
-  maxPurchase1Day?: string;
-  maxPurchaseName?: string;
-  maxPurchaseImgUrl?: string | undefined;
-  maxPurchaseTokenAddress?: string | undefined;
-};
+export type BroadcastMessageEventDetails = Extract<
+  Types.NotificationHistoryEntryFragmentFragment['detail'],
+  { __typename: 'BroadcastMessageEventDetails' }
+>;
+
+export type ChatMessageReceivedEventDetails = Extract<
+  Types.NotificationHistoryEntryFragmentFragment['detail'],
+  { __typename: 'ChatMessageReceivedEventDetails' }
+>;
+
+export type DAOProposalChangedEventDetails = Extract<
+  Types.NotificationHistoryEntryFragmentFragment['detail'],
+  { __typename: 'DAOProposalChangedEventDetails' }
+>;
+
+export type DirectTenantMessageEventDetails = Extract<
+  Types.NotificationHistoryEntryFragmentFragment['detail'],
+  { __typename: 'DirectTenantMessageEventDetails' }
+>;
+
+export type GenericEventDetails = Extract<
+  Types.NotificationHistoryEntryFragmentFragment['detail'],
+  { __typename: 'GenericEventDetails' }
+>;
+
+// With Fallback because it is not queried in gql fragments
+export type GenericEventAction = Extract<
```

Comment:
I think we can actually remove this. i dont see it being used anywhere!

Reply:
yah. .. removed. good point, thanks!!